### ISSUE TITLE
custom_column_names not applied in SQL Query 

### ISSUE THREAD

**Issue by raphaelschwinger** [hasura/graphql-engine#8314]
### Version Information

Server Version:
CLI Version (for CLI related issue): `v2.4.0-beta.1`

### Environment

Docker: `hasura/graphql-engine:v2.2.0`

### What is the expected behaviour?

I added an extisting MSSQL Server Database to Hasura. This Database contains a table named `Article`:
```sql
CREATE TABLE [dbo].[Article](
	[Customer UUID] [uniqueidentifier] NOT NULL,
	[Number] [int] NOT NULL,
	[Name] [nvarchar](max) NULL,
	[Comparison Article] [int] NULL,
	[ABC-Category] [nchar](1) NULL,
	[Shelf Life] [int] NULL,
	[Article Group] [int] NULL,
	[Is Forecast Article] [bit] NOT NULL,
	[Cooling Type] [nvarchar](6) NULL,
 CONSTRAINT [PK_Article] PRIMARY KEY CLUSTERED 
(
	[Customer UUID] ASC,
	[Number] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
GO
```

The table is tracked within hasura resulting in the following metadata:
```
table:
  name: Article
  schema: dbo
configuration:
  custom_column_names:
    ABC-Category: ABC_Category
    Article Group: Article_Group
    Comparison Article: Comparison_Article
    Cooling Type: Cooling_Type
    Customer UUID: Customer_UUID
    Is Forecast Article: Is_Forecast_Article
    Shelf Life: Shelf_Life
  custom_root_fields: {}
```

I expect to be able to run this migration:
```graphql
mutation MyMutation {
  insert_Article(objects: {Is_Forecast_Article: false, Customer_UUID: "1", Number: 12}) {
    returning {
      Customer_UUID
    }
  }
}
```

### Keywords

* custom_column_names
* invalid characters  in this identifier
* underscores

### What is the current behaviour?

It seams that the custom_column_names are not applied "backwards", resulting in the following error:
```
{
  "errors": [
    {
      "extensions": {
        "internal": {
          "exception": {
            "tag": "unsuccessful_return_code",
            "contents": [
              "odbc_SQLExecDirectW",
              -1,
              "[Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid column name 'Cooling_Type'."
            ]
          },
          "query": "SELECT [Cooling_Type], [ABC_Category], [Comparison_Article], [Article_Group], [Name], [Shelf_Life], [Is_Forecast_Article], [Number], [Customer_UUID] INTO #inserted FROM [dbo].[Article] WHERE (1<>1) UNION ALL SELECT [Cooling_Type], [ABC_Category], [Comparison_Article], [Article_Group], [Name], [Shelf_Life], [Is_Forecast_Article], [Number], [Customer_UUID] FROM [dbo].[Article] WHERE (1<>1)"
        },
        "path": "$",
        "code": "unexpected"
      },
      "message": "database query error"
    }
  ]
}
```

### How to reproduce the issue?

1.  add the table as described before
2. track table
3. run graphql mutation in GraphiQL

### Screenshots or Screencast

<img width="1365" alt="Bildschirmfoto 2022-03-13 um 15 13 25" src="https://user-images.githubusercontent.com/27692987/158063760-372b4b7a-2d87-4526-8811-f497e988f67f.png">
<img width="647" alt="Bildschirmfoto 2022-03-13 um 15 13 52" src="https://user-images.githubusercontent.com/27692987/158063769-554d0fd3-1e3a-41b2-8854-e20642fd267b.png">

### Please provide any traces or logs that could help here.

What kind of logs could I provide?
### If the bug is confirmed, would you be willing to submit a PR?

Yes, I would like to help fix this bug.

**Comment by sordina** [comment]
@dmoverton would you be able to confirm the mechanism here? I think we had taken this situation into account, so maybe there is an issue with the generalized joins?

**Comment by dmoverton** [comment]
> @dmoverton would you be able to confirm the mechanism here? I think we had taken this situation into account, so maybe there is an issue with the generalized joins?

@sordina I can reproduce it. Nothing to do with generalized joins - there is only one table in the example. The SQL the insert mutation generates looks weird and definitely is using the wrong fields names. I think this is one for the data sources team.

**Comment by raphaelschwinger** [comment]
Thank you for taking a look into this. I created a [repo](https://github.com/raphaelschwinger/hasura-custom_column_name-bug) to replicate this issue.

Also it is worth mentioning that a query is translated correctly. For example in the test repo:
```graphql
query MyQuery {
  Article {
    Name
    Number
    With_Space
  }
}
```

To:
```sql
SELECT
  ISNULL(
    (
      SELECT
        [t_Article1].[Name] AS [Name],
        [t_Article1].[Number] AS [Number],
        [t_Article1].[With Space] AS [With_Space]
      FROM
        [dbo].[Article] AS [t_Article1] FOR JSON PATH,
        INCLUDE_NULL_VALUES
    ),
    '[]'
  )
```

**Comment by soupi** [comment]
@raphaelschwinger thank you for the high-quality bug report. I have opened a PR fixing this issue.

**Comment by raphaelschwinger** [comment]
@soupi Could you link your PR? We are eager to work with Hasura :)

**Comment by soupi** [comment]
Hi @raphaelschwinger, [the fix has been merged](https://github.com/hasura/graphql-engine/commit/f06bff40081227c63a33c08356147d6db82bb786) and I believe it can be previewed with this docker build: `hasurabuild/graphql-engine:mono-branch-main-c1c4e32c7`

**Comment by soupi** [comment]
Thanks for confirming @raphaelschwinger!
The issue can be closed now, and the fix will be part of the next release.

**Comment by soupi** [comment]
Fixed via f06bff40081227c63a33c08356147d6db82bb786
